Tests/UnitTests/Plugins/ActSVG/IndexedSurfacesSvgConverterTests.cpp

```diff
@@ -0,0 +1,228 @@
+
+  // Hierarchy map of styles
+  GeometryHierarchyMap<Style> surfaceStyles({allSensitives});
+  isOptions.surfaceStyles = surfaceStyles;
+
+  // The grid style
+  GridConverter::Options gridOptions;
+  Style gridStyle;
+  gridStyle.fillOpacity = 0.;
+  gridStyle.strokeColor = {0, 0, 255};
+  gridStyle.strokeWidth = 1.;
+  gridStyle.highlightStrokeWidth = 3;
+  gridStyle.highlightStrokeColor = {255, 0, 0};
+  gridOptions.style = gridStyle;
+
+  isOptions.gridOptions = gridOptions;
+  return isOptions;
+};
+
+auto drawOptions = generateDrawOptions();
+
+BOOST_AUTO_TEST_SUITE(DetectorSvg)
+
+BOOST_AUTO_TEST_CASE(RingDisc1D) {
+  // A single ring
+  CylindricalTrackingGeometry::DetectorStore dStore;
+  auto rSurfaces = cGeometry.surfacesRing(dStore, 6.4, 12.4, 36., 0.125, 0.,
+                                          55., 0., 2., 22u);
+
+  IndexedSurfacesGenerator<decltype(rSurfaces)> irSurfaces{
+      rSurfaces, {}, {binPhi}};
+
+  GridAxisGenerators::EqClosed aGenerator{{-M_PI, M_PI}, 44u};
+  PolyhedronReferenceGenerator rGenerator{true, 1};
+
+  auto indexedRing = irSurfaces(tContext, aGenerator, rGenerator);
+  // The displaying
+  auto pIndexedRing = IndexedSurfacesConverter::convert(
+      tContext, rSurfaces, indexedRing, drawOptions);
+  auto pIndexRingView = View::xy(pIndexedRing, "RingDisc1D");
+  toFile({pIndexRingView}, pIndexRingView._id + ".svg");
+}
+
+BOOST_AUTO_TEST_CASE(RingDisc1DWithSupport) {
+  // A single ring
+  CylindricalTrackingGeometry::DetectorStore dStore;
+  auto rSurfaces = cGeometry.surfacesRing(dStore, 6.4, 12.4, 36., 0.125, 0.,
+                                          55., 0., 2., 22u);
+
+  auto rBounds = std::make_shared<RadialBounds>(20., 20.);
+  auto dSurface = Surface::makeShared<DiscSurface>(Transform3::Identity(),
+                                                   std::move(rBounds));
+  rSurfaces.push_back(dSurface.get());
+
+  IndexedSurfacesGenerator<decltype(rSurfaces)> irSurfaces{
+      rSurfaces, {rSurfaces.size() - 1u}, {binPhi}};
+
+  GridAxisGenerators::EqClosed aGenerator{{-M_PI, M_PI}, 44u};
+  PolyhedronReferenceGenerator rGenerator{true, 1};
+
+  auto indexedRing = irSurfaces(tContext, aGenerator, rGenerator);
+  // The displaying
+  auto pIndexedRing = IndexedSurfacesConverter::convert(
+      tContext, rSurfaces, indexedRing, drawOptions);
+  auto pIndexRingView = View::xy(pIndexedRing, "RingDisc1DWithSupport");
+  toFile({pIndexRingView}, pIndexRingView._id + ".svg");
+}
+
+BOOST_AUTO_TEST_CASE(RingDisc2D) {
+  // Two rings to make a disc
+  CylindricalTrackingGeometry::DetectorStore dStore;
+  auto rSurfacesR0 = cGeometry.surfacesRing(dStore, 6.4, 12.4, 18., 0.125, 0.,
+                                            42., 0., 2., 22u);
+
+  auto rSurfacesR1 = cGeometry.surfacesRing(dStore, 12.4, 20.4, 30., 0.125, 0.,
+                                            80., 0., 2., 22u);
+
+  decltype(rSurfacesR0) rSurfaces = rSurfacesR0;
```

Comment:
Why is this copy needed? ```rSurfacesR0``` does not seem to be used otherwise

Reply:
Here, we are using the already existing infrastructure of the 'CylindricalSurfaceGeometry' which returns a `const vector` - hence we copy, to add the other surfaces.

Reply:
Ah, Ok, I could not see the ```const```